src/cbor/CBOREncoder.cpp

```diff
@@ -45,24 +46,68 @@ CborError CBOREncoder::encode(PropertyContainer & property_container, uint8_t *
    */
   CborError error = CborNoError;
   int num_encoded_properties = 0;
-  std::for_each(property_container.begin(),
-                property_container.end(),
-                [lightPayload, &arrayEncoder, &error, &num_encoded_properties](Property * p)
-                {
-                  if (p->shouldBeUpdated() && p->isReadableByCloud())
-                  {
-                    error = p->append(&arrayEncoder, lightPayload);
-                    if(error == CborNoError)
-                      num_encoded_properties++;
-                    else
-                      return;
-                  }
-                });
-  if ((CborNoError != error) && 
-      (CborErrorOutOfMemory != error))
+  int num_checked_properties = 0;
+  static int encoded_properties_message_limit = CBOR_ENCODER_NO_PROPERTIES_LIMIT;
+
+  if(current_property_index >= property_container.size())
+    current_property_index = 0;
+
+  PropertyContainer::iterator iter = property_container.begin();
+  std::advance(iter, current_property_index);
+
+  for(; iter != property_container.end(); iter++)
+  {
+    Property * p = * iter;
+    bool maximum_number_of_properties_reached = (num_encoded_properties >= encoded_properties_message_limit) && (encoded_properties_message_limit != -1);
+    bool cbor_encoder_error = (error != CborNoError);
```

Comment:
```suggestion
    bool const cbor_encoder_error = (error != CborNoError);
```
Same as above. Btw, I love it that you are "encoding" the meaning of a boolean operation in the name of the boolean variable. Makes the if statement below so much more easily readable.

Reply:
Same as above also fo me.